tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator/Targets/Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator.targets

```diff
@@ -40,6 +43,12 @@
           Condition="$(_IsFunctionsSdkBuild) != 'true'"
           DependsOnTargets="RunResolvePublishAssemblies">
 
+    <PropertyGroup>
+      <_IsFunctionsSdkBuild Condition="$(_FunctionsTaskFramework) != ''">true</_IsFunctionsSdkBuild>
+      <_FunctionsExtensionsDir>$(TargetDir)</_FunctionsExtensionsDir>
+      <_FunctionsExtensionsDir Condition="$(_IsFunctionsSdkBuild) == 'true'">$(_FunctionsExtensionsDir)bin</_FunctionsExtensionsDir>
+    </PropertyGroup>
```

Comment:
I don't think moving the `_IsFunctionsSdkBuild` evaluation inside the targets will work because the condition on this Target wouldn't be properly evaluated yet. 

I think the issue really lives in https://github.com/Azure/azure-functions-vs-build-sdk/issues/277, where we need to make sure the ordering is consistent.

Reply:
Yep, bringing in `_IsFunctionsSdkBuild` within the `_GenerateFunctionsExtensionsMetadataPostBuild` target does solve the initial problem with registering extensions. 

But it does miss the fairly obvious recursion it introduces around the `_ResolveAdditionalReferenceCopyLocalPathsForFunctionsExtensions` target.